packages/react-components/src/components/VideoGallery/SpeakerVideoLayout.tsx

```diff
@@ -0,0 +1,249 @@
+import {
+  SMALL_FLOATING_MODAL_SIZE_REM,
+  LARGE_FLOATING_MODAL_SIZE_REM,
+  localVideoTileContainerStyle
+} from './styles/FloatingLocalVideo.styles';
+/* @conditional-compile-remove(vertical-gallery) */
+import {
+  VERTICAL_GALLERY_FLOATING_MODAL_SIZE_REM,
+  SHORT_VERTICAL_GALLERY_FLOATING_MODAL_SIZE_REM
+} from './styles/FloatingLocalVideo.styles';
+import { useOrganizedParticipants } from './utils/videoGalleryLayoutUtils';
+import { GridLayout } from '../GridLayout';
+import { rootLayoutStyle } from './styles/FloatingLocalVideoLayout.styles';
+import { layerHostStyle, innerLayoutStyle } from './styles/FloatingLocalVideoLayout.styles';
+import { videoGalleryLayoutGap } from './styles/Layout.styles';
+import { useId } from '@fluentui/react-hooks';
+
+/**
+ * Props for {@link FloatingLocalVideoLayout}.
+ *
+ * @private
+ */
+export interface SpeakerVideoLayoutProps extends LayoutProps {
+  /**
+   * Whether to display the local video camera switcher button
+   */
+  showCameraSwitcherInLocalPreview?: boolean;
+  /**
+   * Height of parent element
+   */
+  parentHeight?: number;
+  /* @conditional-compile-remove(click-to-call) */
+  /**
+   * Local video tile mode
+   */
+  localVideoTileSize?: LocalVideoTileSize;
+}
+
+/**
+ * Layout for the gallery mode to highlight the current dominant speaker
+ *
+ * @private
+ */
+export const SpeakerVideoLayout = (props: SpeakerVideoLayoutProps): JSX.Element => {
+  const {
+    remoteParticipants = [],
+    dominantSpeakers,
+    localVideoComponent,
+    screenShareComponent,
+    onRenderRemoteParticipant,
+    styles,
+    maxRemoteVideoStreams,
+    parentWidth,
+    /* @conditional-compile-remove(vertical-gallery) */ parentHeight,
+    /* @conditional-compile-remove(vertical-gallery) */ overflowGalleryPosition = 'HorizontalBottom',
+    pinnedParticipantUserIds = [],
+    /* @conditional-compile-remove(click-to-call) */ localVideoTileSize
+  } = props;
+
+  const theme = useTheme();
+
+  const isNarrow = parentWidth ? isNarrowWidth(parentWidth) : false;
+
+  /* @conditional-compile-remove(vertical-gallery) */
+  const isShort = parentHeight ? isShortHeight(parentHeight) : false;
+
+  // This is for tracking the number of children in the first page of overflow gallery.
+  // This number will be used for the maxOverflowGalleryDominantSpeakers when organizing the remote participants.
+  const childrenPerPage = useRef(4);
+  const { gridParticipants, overflowGalleryParticipants } = useOrganizedParticipants({
+    remoteParticipants,
+    dominantSpeakers,
+    maxRemoteVideoStreams,
+    isScreenShareActive: !!screenShareComponent,
+    maxOverflowGalleryDominantSpeakers: screenShareComponent
+      ? childrenPerPage.current - (pinnedParticipantUserIds.length % childrenPerPage.current)
+      : childrenPerPage.current,
+    /* @conditional-compile-remove(pinned-participants) */ pinnedParticipantUserIds,
+    /* @conditional-compile-remove(gallery-layouts) */ layout: 'speaker'
```

Comment:
I'm totally fine with adding a layout prop to useOrganizedParticipants but I am curious if you can instead set maxRemoteVideoStreams as 0 and set remoteParticipants as `remoteParticipants.filter(p => userId !== dominantSpeaker[0])` and gridTiles below (on line 97) would just be `remoteParticipants.find(p => userId === dominantSpeaker[0])`. Even if this works, it may be less intuitive for code readers

Reply:
That I think would work, but I like the use of the layout to do this determination, helps understand why we do or don't do something other than just the filtering of participants.